lib/Conversion/ImportVerilog/Expressions.cpp

```diff
@@ -791,6 +792,66 @@ struct RvalueExprVisitor {
     return visitAssignmentPattern(expr, *count);
   }
 
+  Value visit(const slang::ast::StreamingConcatenationExpression &expr) {
+    SmallVector<Value> operands;
+    for (auto stream : expr.streams()) {
+      auto operandLoc = context.convertLocation(stream.operand->sourceRange);
+      if (stream.constantWithWidth.has_value() || stream.withExpr) {
+        mlir::emitError(operandLoc) << "Moore only support streaming "
+                                       "concatenation without 'with operation'";
+        return {};
+      }
+
+      auto value = context.convertRvalueExpression(*stream.operand);
+      if (!value)
+        return {};
+      value = context.convertToSimpleBitVector(value);
+      if (!value) {
+        return {};
+      }
+      operands.push_back(value);
+    }
+    Value value;
+
+    if (operands.size() == 1) {
+      // there must be at least one element, otherwise slang will report an
+      // error
+      value = operands.front();
+    } else {
+      value = builder.create<moore::ConcatOp>(loc, operands).getResult();
+    }
+
+    if (expr.sliceSize == 0) {
+      return value;
+    }
+
+    auto type = cast<moore::IntType>(value.getType());
+    SmallVector<Value> slicedOperands;
+    auto iterMax = type.getWidth() / expr.sliceSize;
+    auto remainSize = type.getWidth() % expr.sliceSize;
+
+    for (size_t i = 0; i < iterMax; i++) {
+      auto extractResultType = moore::IntType::get(
+          context.getContext(), expr.sliceSize, type.getDomain());
+
+      auto extracted = builder.create<moore::ExtractOp>(
+          loc, extractResultType, value, i * expr.sliceSize);
+      slicedOperands.push_back(extracted);
+    }
+    // handle other wire
+    if (remainSize) {
+      auto extractResultType = moore::IntType::get(
+          context.getContext(), remainSize, type.getDomain());
+
+      auto extracted = builder.create<moore::ExtractOp>(
+          loc, extractResultType, value, iterMax * expr.sliceSize);
+      slicedOperands.push_back(extracted);
+    }
+
+    std::reverse(slicedOperands.begin(), slicedOperands.end());
```

Comment:
From my perspective, we needn't reverse the `slicedOperands`. In your case:
```
// CHECK:     %[[VAL_23:.*]] = moore.constant -11 : i6
// CHECK:     %[[VAL_24:.*]] = moore.extract %[[VAL_23]] from 0 : i6 -> i4
// CHECK:     %[[VAL_25:.*]] = moore.extract %[[VAL_23]] from 4 : i6 -> i2
// CHECK:     %[[VAL_26:.*]] = moore.concat %[[VAL_25]], %[[VAL_24]] : (!moore.i2, !moore.i4) -> i6
a = { << 4 { 6'b11_0101 }};
```
The `<< 4 {6'b11_0101}` returns `0101_11`. `%[[VAL_24:.*]]` is `0101`, and `%[[VAL_25:.*]]` is `11`. So `moore.concat [VAL_24], [VAL_25]` is right.